backend/copr_backend/pulp.py

```diff
@@ -178,3 +189,57 @@ def delete_distribution(self, distribution):
         """
         url = self.config["base_url"] + distribution
         return requests.delete(url, **self.request_params)
+
+    def list_packages(self, repository_version):
+        """
+        List packages for a given repository version
+        Ideally, we would list packages for a repository but that probably
+        isn't possible.
+
+        TODO Since we don't care about repository versions, we should probably
+        set `--retain-repo-versions=1`
+        https://discourse.pulpproject.org/t/delete-artifact-with-pulp-cli/559/6
+
+        https://pulpproject.org/pulp_rpm/restapi/#tag/Content:-Packages/operation/content_rpm_packages_list
+        """
+        url = self.url("api/v3/content/rpm/packages/?")
+        url += urlencode({"repository_version": repository_version})
+        return requests.get(url, **self.request_params)
+
+    def list_repositories(self, prefix):
+        """
+        Get a list of repositories whose names match a given prefix
+        https://pulpproject.org/pulp_rpm/restapi/#tag/Repositories:-Rpm/operation/repositories_rpm_rpm_list
+        """
+        url = self.url("api/v3/repositories/rpm/rpm/?")
+        url += urlencode({"name__startswith": prefix})
+        return requests.get(url, **self.request_params)
+
+    def get_latest_repository_version(self, name):
+        """
+        Return the latest repository version
+        https://pulpproject.org/pulp_rpm/restapi/#tag/Repositories:-Rpm-Versions/operation/repositories_rpm_rpm_versions_list
+        """
+        response = self.get_repository(name)
+        href = response.json()["results"][0]["versions_href"]
+
+        url = self.config["base_url"] + href + "?"
+        url += urlencode({"offset": 0, "limit": 1})
+
+        response = requests.get(url, **self.request_params)
+        return response.json()["results"][0]["pulp_href"]
+
+    def wait_for_finished_task(self, task):
```

Comment:
No idea to be honest, if a task can get stuck on the Pulp side. I think we will get stuck in this loop "forever". But "forever" should be only until Pulp gets unstuck.

Should I add some timeout on our side?

Reply:
I'd say rather yes, since some rare errors may happen and then we could end up here again https://github.com/fedora-copr/copr/issues/3343. Let's just put there some unlikely high timeout - e.g. 5 hours.

Reply:
Done